frontend/packages/kubevirt-plugin/src/components/vm-templates/vm-template-source.tsx

```diff
@@ -0,0 +1,97 @@
+import * as React from 'react';
+import classNames from 'classnames';
+import { TemplateKind } from '@console/internal/module/k8s';
+import { parseURL, resolveURL } from '../../utils/url';
+import { ProvisionSource } from '../../constants/vm/provision-source';
+
+export const URLObj: React.FC<URLObjProps> = ({
+  urlPath,
+  short = false,
+  maxHostnameParts = 3,
+  maxPathnameParts = 1,
+}) => {
+  const urlObj = short ? parseURL(urlPath) : undefined;
+
+  const resolvedURL = urlObj ? resolveURL({ urlObj, maxHostnameParts, maxPathnameParts }) : urlPath;
+  const resolvedTitle = resolvedURL === urlPath ? undefined : urlPath;
+
+  return (
+    <a href={urlPath} title={resolvedTitle}>
+      {resolvedURL}
+    </a>
+  );
+};
+
+type URLObjProps = {
+  urlPath: string;
+  short?: boolean;
+  maxHostnameParts?: number;
+  maxPathnameParts?: number;
+};
+
+const Type: React.FC<TypeProps> = ({ type, source, error, isInline = false }) => (
+  <div
+    title={source || error}
+    className={classNames({ 'kubevirt-template-source__overlay': isInline })}
+  >
+    {type}
+  </div>
+);
+
+type TypeProps = {
+  type: string;
+  source?: string;
+  error?: string;
+  isInline?: boolean;
+};
+
+const Source: React.FC<SourceProps> = ({ type, source }) => {
+  if (!source) {
+    return null;
+  }
+
+  const sourceElem =
+    type === ProvisionSource.URL.getValue() ? <URLObj urlPath={source} short /> : source;
+
+  return <div className="kubevirt-template-source__source">{sourceElem}</div>;
+};
+
+type SourceProps = {
+  type: string;
+  source?: string;
+};
+
+export const TemplateSource: React.FC<TemplateSourceProps> = ({
+  template,
+  detailed = false,
+  showError = false,
+}) => {
+  const provisionSource = ProvisionSource.getProvisionSourceDetails(template);
+  const { type, source, error } = provisionSource;
+
+  if (!provisionSource || !provisionSource.type) {
+    if (showError) {
```

Comment:
still, `'No bootable device found.'` error is the same as `Not available`. I would rather use just the later one to be consistent

Reply:
Just to be clear, do you want: `No bootable device found` or `Not available` error message ?

Reply:
`Not available` should be fine